utils/pool_test.go

```diff
@@ -0,0 +1,25 @@
+package utils
+
+import (
+	"sync/atomic"
+	"testing"
+)
+
+func TestWaitGroupPool(t *testing.T) {
+	wgp := NewWaitGroupPool(10)
+
+	var total uint32
+
+	for i := 0; i < 1000; i++ {
```

Comment:
maybe 100? Although go is fast, we don't need unnecessary loop

Reply:
done!